File: homework/dfs.c
#include <stdio.h>
#include <string.h>

int visited[9];
int graph[9][3] = {{}, {2,3,8}, {1,7}, {1,4,5}, {3,5}, {3,4}, {7}, {2,6,8}, {1,7}};

void dfs(int x) {
    visited[x] = 1;
    printf("%d ", x);
    int nodeLength = sizeof(graph[x]) / sizeof(int);
    for (int i = 0; i < nodeLength; i++) {
        int node = graph[x][i];
        // 인접한 노드가 방문한 적이 없다면 DFS 수행
        if(node == 0) continue;
        
        if(!visited[node]) {    
            dfs(node);
        }
    }
}

int main(void) {
    dfs(1);
    return 0;
}

File: java/algo_java/src/dfs/Dfs.java
package dfs;

public class Dfs {
    public static boolean[] visited = new boolean[9];
    static int [][] graph =  {{}, {2,3,8}, {1,7}, {1,4,5}, {3,5}, {3,4}, {7}, {2,6,8}, {1,7}};

    // DFS함수 정의
    public static void dfs(int x) {
        // 현재 노드를 방문 처리
        visited[x] = true;
        System.out.println(x + " ");
        // 현재 노드와 연결된 다른 노드를 재귀적으로 방문
        for(int node : graph[x]) {
            if(!visited[node]) {
                dfs(node);
            }
        }
    }

    public static void main(String[] args) {
        dfs(1);
    }
}
